### PR TITLE
Resolve the flakiness of EmitsSearchTelemetryEvents by making it fully offline

### LINKED ISSUE #8393: Test:  NuGet.PackageManagement.UI.Test.PackageItemLoaderTests.EmitsSearchTelemetryEvents is flaky
Test `NuGet.PackageManagement.UI.Test.PackageItemLoaderTests.EmitsSearchTelemetryEvents` is flaky.
```
NuGet.PackageManagement.UI.Test.PackageItemLoaderTests.EmitsSearchTelemetryEvents [FAIL]
  Assert.Equal() Failure
  Expected: 4
  Actual:   5
  Stack Trace:
    C:\git\NuGet.Client\test\NuGet.Clients.Tests\NuGet.PackageManagement.UI.Test\PackageItemLoaderTests.cs(107,0): at NuGet.PackageManagement.UI.Test.PackageItemLoaderTests.<EmitsSearchTelemetryEvents>d__2.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/8393
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 
The package item loader has a short 5s timeout before deciding to update the results. 

In this test case, we try to load 2 pages. 
The way we load pages is by looking at the previous result and just offsetting as required.
In the off case, that the execution of this remote call (read load) takes longer, we'll end up sending 2 search events as there's no continuation token from the previous operation.

This should hopefully ensure that the test completes in the expected time :) 

fyi @joelverhagen 

## Testing/Validation

Tests Added: Yes/No  
Reason for not adding tests:  
Validation:

### CODE REVIEW

File: test/TestUtilities/Test.Utility/SourceRepository/TestSourceRepositoryUtility.cs [hunk 1]
```diff
@@ -52,6 +53,18 @@ public static SourceRepositoryProvider CreateSourceRepositoryProvider(IEnumerabl
             return sourceRepositoryProvider;
         }
 
+        public static SourceRepositoryProvider CreateSourceRepositoryProvider(IEnumerable<PackageSource> packageSources, IEnumerable<Lazy<INuGetResourceProvider>> mockNuGetResourceProviders)
+        {
+            var thisUtility = new TestSourceRepositoryUtility();
+            //var container = thisUtility.Initialize();
```
Comment (zivkan) [inline test/TestUtilities/Test.Utility/SourceRepository/TestSourceRepositoryUtility.cs]: no commented out code please

**Review by zivkan — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/PackageItemLoaderTests.cs [hunk 2]
```diff
@@ -88,9 +90,30 @@ public async Task EmitsSearchTelemetryEvents()
...
+                    query + "?q=nuget&skip=0&take=26&prerelease=true&semVerLevel=2.0.0",
+                    ProtocolUtility.GetResource("NuGet.PackageManagement.UI.Test.compiler.resources.nugetSearchPage1.json", typeof(PackageItemLoaderTests))
+                },
+                {
+                    query + "?q=nuget&skip=25&take=26&prerelease=true&semVerLevel=2.0.0",
+                    ProtocolUtility.GetResource("NuGet.PackageManagement.UI.Test.compiler.resources.nugetSearchPage2.json", typeof(PackageItemLoaderTests))
+                },
+            };
+
+            var httpSource = new TestHttpSource(new PackageSource(source), responses);
```
Comment (dtivel) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/PackageItemLoaderTests.cs]: `TestHttpSource` is disposable.

**Review by dtivel — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/compiler/resources/nugetSearchPage1.json [hunk 3]
```diff
@@ -0,0 +1,5320 @@
...
+          "@id": "https://api.nuget.org/v3/registration3-gz-semver2/nuget.versioning/4.8.0-preview1.5156.json"
+        },
+        {
+          "version": "4.8.0-preview3.5278+c3240b16fcf3276246fc8c610771d14ab94fdc02",
+          "downloads": 282,
+          "@id": "https://api.nuget.org/v3/registration3-gz-semver2/nuget.versioning/4.8.0-preview3.5278.json"
+        },
+        {
+          "version": "4.8.0-rtm.5362+dc0b55f597c6224a13d64b984651b18700d3054a",
+          "do
```
Comment (loic-sharma) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/compiler/resources/nugetSearchPage1.json]: Can you use fake data instead? We shouldn't store customer's data like this due to GDPR

Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/compiler/resources/nugetSearchPage1.json]: Discussed offline, I'll patch it up :)

File: test/TestUtilities/Test.Utility/SourceRepository/TestSourceRepositoryUtility.cs (hunk 1 above)
Comment (nkolev92) [reply test/TestUtilities/Test.Utility/SourceRepository/TestSourceRepositoryUtility.cs]: Already did that but didn't commit. 

Fixed it again.

**Pull request merged**